src/api/environment.cc

```diff
@@ -1004,6 +1007,11 @@ void DefaultProcessExitHandlerInternal(Environment* env, ExitCode exit_code) {
   // in node_v8_platform-inl.h
   uv_library_shutdown();
   DisposePlatform();
```

Comment:
I noticed that in the normal exit path, this is done separately and guarded by kNoInitializeNodeV8Platform 

https://github.com/nodejs/node/blob/ca76b393561e0bdb19818407e6ec950a97d714b0/src/node.cc#L1312-L1321

Maybe we should wrap the things that need to be done both on normal and abnormal exit into a helper and call them in both DefaultProcessExitHandlerInternal and TearDownOncePerProcess?

(Also, from what I can tell, this function would only get called on the main thread - it would be better to assert it, most things in this section can only be called once by one thread)